src/custom/state/claim/hooks/index.ts

```diff
@@ -1061,3 +1066,71 @@ export function useVCowData(): VCowData {
 
   return { isLoading, vested, unvested, total }
 }
+
+/**
+ * Hook used to swap vCow to Cow token
+ */
+
+interface SwapVCowCallbackParams {
+  openModal: (message: string, operationType: OperationType) => void
+  closeModal: () => void
+}
+
+export function useSwapVCowCallback({ openModal, closeModal }: SwapVCowCallbackParams) {
+  const { chainId, account } = useActiveWeb3React()
+  const vCowContract = useVCowContract()
+
+  const addTransaction = useTransactionAdder()
+  const vCowToken = chainId ? V_COW[chainId] : undefined
+
+  const swapCallback = useCallback(async () => {
+    if (!account) {
+      throw new Error('Not connected')
+    }
+    if (!chainId) {
+      throw new Error('No chainId')
+    }
+    if (!vCowContract) {
+      throw new Error('vCOW contract not present')
+    }
+    if (!vCowToken) {
+      throw new Error('vCOW token not present')
+    }
+
+    const args: { from: string; gasLimit?: BigNumber } = {
+      from: account,
```

Comment:
nit: are you sure we want to include this? I thought in the call we mentioned this was not needed (approve and wrap don't have it and is implied in the wallet)

Reply:
My understanding was that we want to add this because some wallets calculate this badly

Reply:
> My understanding was that we want to add this because some wallets calculate this badly

Why do you say this? Also if this is the case, why aren't we doing it in the wrapping an approve?

Reply:
It is there @anxolin  😂

https://github.com/gnosis/cowswap/blob/baf229374a68c207f1efbb1fb0279bc8a630405a/src/custom/hooks/useApproveCallback/useApproveCallbackMod.ts#L128